### PR TITLE
Add limits for parameter, return_value, and logs

### PR DESCRIPTION
**Pull request by DOBEN**
## Purpose

Add the limits to documentation regarding the parameter size limit, return_value size limit, and number of log items.

Note:  This should not be merged before the protocol update is in effect on testnet.

### CODE REVIEW

File: source/mainnet/smart-contracts/general/contract-instances.rst [hunk 1]
```diff
@@ -152,6 +160,15 @@ reduce cost.
...
 .. _contract-instance-operations:
 
+Limits
+======
+
+When initializing or invoking a smart contract, the following limits are enforced:
+
+- The maximum parameter size is 65535 Bytes.
+- Up to 4 billion log items per execution.
+- The maximum return value size is 4 GigaBytes.
```
Comment (Bargsteen) [inline source/mainnet/smart-contracts/general/contract-instances.rst:170]: I am not sure that we should mention these limits as you will hit the energy limit well before that. At least we should mention the energy limit.

File: source/mainnet/smart-contracts/general/contract-instances.rst [hunk 2]
```diff
@@ -152,6 +160,15 @@ reduce cost.
 
 .. _contract-instance-operations:
 
+Limits
+======
+
+When initializing or invoking a smart contract, the following limits are enforced:
```
Comment (Bargsteen) [inline source/mainnet/smart-contracts/general/contract-instances.rst:166]:
```suggestion
When initializing, updating, or invoking a smart contract, the following limits are enforced:
```

**Review by Bargsteen — CHANGES_REQUESTED** [review]

File: source/mainnet/smart-contracts/general/contract-instances.rst (hunk 1 above)
Comment (DOBEN) [reply source/mainnet/smart-contracts/general/contract-instances.rst:170]: What is the current energy limit for the init/receive function?

**Review by dg-concordium — APPROVED** [review]

File: source/mainnet/smart-contracts/general/contract-instances.rst [hunk 3]
```diff
@@ -152,6 +160,15 @@ reduce cost.
...
+
+- The maximum parameter size is 65535 Bytes.
+- Up to 4 billion log items per execution.
```
Comment (abizjak) [inline source/mainnet/smart-contracts/general/contract-instances.rst:169]: We can be more precise here, it is `u32::MAX` logs.

File: source/mainnet/smart-contracts/general/contract-instances.rst (hunk 1 above)
Comment (abizjak) [inline source/mainnet/smart-contracts/general/contract-instances.rst:170]: To be more precise it is u32::MAX bytes. 

I think we should have the limits, but still state that they cannot be reached in practice due to NRG bounds.

**Review by abizjak — APPROVED** [review]

File: source/mainnet/smart-contracts/general/contract-instances.rst [hunk 4]
```diff
@@ -152,6 +160,20 @@ reduce cost.
...
 
+Limits
+======
+
+When initializing, updating, or invoking a smart contract, the following limits are enforced:
+
+- The maximum input parameter size is 65535 Bytes (u16::MAX).
+- The energy limit in a block is 3 million NRG (energy). If only one transaction is in the block, the transaction can consume up to 3 million NRG (energy).
+- The maximim number of log items is u32::MAX per execution.
+- The maximum return value size is u32::MAX Bytes.
```
Comment (Bargsteen) [inline source/mainnet/smart-contracts/general/contract-instances.rst:171]:
```suggestion
- The maximum input parameter size is 65535 Bytes (``u16::MAX``).
- The energy limit in a block is 3 million NRG (energy). If only one transaction is in the block, the transaction can consume up to 3 million NRG (energy).
- The maximum number of log items is ``u32::MAX`` per execution.
- The maximum return value size is ``u32::MAX`` Bytes.
```

**Review by Bargsteen — APPROVED** [review]
LGTM. But should not be merged before the protocol update is in effect on testnet.

**Pull request merged**